packages/core/content-manager/server/src/services/entity-manager.ts

```diff
@@ -180,31 +241,22 @@ export default ({ strapi }: { strapi: Strapi }) => ({
     return strapi.entityService.deleteMany(uid, opts);
   },
 
-  async publish(entity: Entity, uid: Common.UID.ContentType, body = {}) {
-    if (entity[PUBLISHED_AT_ATTRIBUTE]) {
-      throw new ApplicationError('already.published');
-    }
-
-    // validate the entity is valid for publication
-    await strapi.entityValidator.validateEntityCreation(
-      strapi.getModel(uid),
-      entity,
-      undefined,
-      // @ts-expect-error - FIXME: entity here is unnecessary
-      entity
-    );
-
-    const data = { ...body, [PUBLISHED_AT_ATTRIBUTE]: new Date() };
+  async publish(
+    document: Entity,
+    uid: Common.UID.ContentType,
+    opts: Parameters<Documents.RepositoryInstance['publish']>[1] = {} as any
+  ) {
     const populate = await buildDeepPopulate(uid);
+    const params = { ...opts, populate };
 
-    const params = { data, populate };
-
-    const updatedEntity = await strapi.entityService.update(uid, entity.id, params);
-
-    await emitEvent(uid, ENTRY_PUBLISH, updatedEntity!);
+    const { versions: publishedDocuments } = await strapi
+      .documents(uid)
+      // @ts-expect-error - Change entity to document
+      .publish(document.id, params);
 
-    // @ts-expect-error mapEntity can be extended
-    const mappedEntity = await this.mapEntity(updatedEntity, uid);
+    // TODO: What if we publish many versions at once?
```

Comment:
IMO we either return the default locale unless they passed one in, or nothing at all.